Automattic/wp-calypso
# Plugins: Reduxify plugin updates and autoupdates

We have several plugins stores, and they're pretty tightly coupled with a lot of functionality. They're also some of the most complex flux stores that remain. In #24180 we're working on reduxifying them step by step. This incremental approach is required because migration in a single PR is not a feasible task.

This PR reduxifies plugin updates and autoupdates actions. In order to keep the store and only reduxify the flux actions, it updates the redux action creators to also dispatch the corresponding flux actions and updates all the component flux action usage with redux.

As a next step, we can start reduxifying the corresponding flux store functionality for updates and autoupdates.

See #47711 for similar work we did for plugin activation and deactivation.

This is part of #24180.

#### Changes proposed in this Pull Request

* Trigger plugin update, autoupdate (enabling and disabling) and autoupdate toggling flux actions from Redux.
* Use plugin update, autoupdate (enabling and disabling) and autoupdate toggling Redux actions instead of flux ones.
* Plugins: Delete obsolete plugin update, autoupdate (enabling and disabling) and autoupdate toggling flux actions.
* Update README to discourage future usage of the old plugins flux store.

#### Testing instructions

* Get a Jetpack site with some plugins for testing.
* Verify updating, enabling autoupdates, disabling autoupdates, and toggling autoupdates of plugins still work everywhere:
  * `/plugins/:plugin/:site`
  * `/plugins/:plugin`
  * `/plugins`
  * `/plugins/manage/:site` - for a single plugin
  * `/plugins/manage/:site` - for multiple selected plugins (bulk actions are enabled by clicking the "Edit All" button)
* Make sure to test two specific scenarios:
  * Install an older version of a plugin, and enable autoupdates. Verify it gets autoupdated immediately.
  * Enable autoupdates for a plugin. Then manually downgrade it to an older version. Verify it gets autoupdated immediately once you load plugins on  `/plugins/manage/:site`
* Verify all tests still pass.

#### Notes
* To download an old version of a plugin, head to its page on https://wordpress.org/plugins/ then copy its link from the download button, and replace the latest tag with an older one (you can see those in the development log of every plugin - https://plugins.trac.wordpress.org/log/:plugin_slug).
* Linting errors are expected - there are a couple of them in a file we touched. I'm not fixing them here to avoid making the PR extra large.
